internal/operators/cnv/cnv_operator.go

```diff
@@ -0,0 +1,117 @@
+func (o *operator) GetHostValidationID() string {
+	return string(models.HostValidationIDCnvRequirementsSatisfied)
+}
+
+// ValidateCluster always return "valid" result
+func (o *operator) ValidateCluster(ctx context.Context, cluster *common.Cluster) (api.ValidationResult, error) {
+	// No need to validate cluster because it will be validate on per host basis
+	return api.ValidationResult{Status: api.Success, ValidationId: o.GetClusterValidationID()}, nil
+}
+
+// ValidateHost always return "valid" result
+func (o *operator) ValidateHost(ctx context.Context, cluster *common.Cluster, host *models.Host) (api.ValidationResult, error) {
+	var inventory models.Inventory
+	if err := json.Unmarshal([]byte(host.Inventory), &inventory); err != nil {
+		o.log.Errorf("Failed to get inventory from host with id %s", host.ID)
+		return api.ValidationResult{Status: api.Failure}, err
+	}
+	// TODO: validate available devices on worker node like gpu and sr-iov and check whether there is enough memory to support them
+	if host.Role == models.HostRoleWorker {
+		cpu, _ := o.GetCPURequirementForWorker(ctx, cluster)
+		if inventory.CPU.Count < cpu {
+			return api.ValidationResult{Status: api.Failure, Reasons: []string{"Insufficient CPU to deploy CNV. Required CPU count is: " + strconv.FormatInt(inventory.CPU.Count, 10)}}, nil
+		}
+		mem, _ := o.GetMemoryRequirementForWorker(ctx, cluster)
+		if inventory.Memory.UsableBytes < mem {
+			return api.ValidationResult{Status: api.Failure, Reasons: []string{"Insufficient memory to deploy CNV. Required memory is: " + strconv.FormatInt(inventory.Memory.UsableBytes, 10) + " bytes"}}, nil
+		}
+	} else if host.Role == models.HostRoleMaster {
+		cpu, _ := o.GetCPURequirementForMaster(ctx, cluster)
+		if inventory.CPU.Count < cpu {
+			return api.ValidationResult{Status: api.Failure, Reasons: []string{"Insufficient CPU to deploy CNV. Required CPU count is: " + strconv.FormatInt(inventory.CPU.Count, 10)}}, nil
+		}
+		mem, _ := o.GetMemoryRequirementForMaster(ctx, cluster)
+		if inventory.Memory.UsableBytes < mem {
+			return api.ValidationResult{Status: api.Failure, Reasons: []string{"Insufficient memory to deploy CNV. Required memory is: " + strconv.FormatInt(inventory.Memory.UsableBytes, 10) + " bytes"}}, nil
+		}
+	}
+	return api.ValidationResult{Status: api.Success}, nil
+}
+
+// GetCPURequirementForWorker provides worker CPU requirements for the operator
+func (o *operator) GetCPURequirementForWorker(context.Context, *common.Cluster) (int64, error) {
+	return 2, nil
+}
+
+// GetCPURequirementForMaster provides master CPU requirements for the operator
+func (o *operator) GetCPURequirementForMaster(context.Context, *common.Cluster) (int64, error) {
+	return 4, nil
+}
+
+// GetMemoryRequirementForWorker provides worker memory requirements for the operator
+func (o *operator) GetMemoryRequirementForWorker(ctx context.Context, cluster *common.Cluster) (int64, error) {
+	return hardware.MibToBytes(360), nil
```

Comment:
Do you think it's worth capturing these values in a structure of its own so that they can be overwritten at runtime via environment variables, like it is being done in [OCS](https://github.com/openshift/assisted-service/blob/master/internal/operators/ocs/validations.go#L30-L45)? To me makes sense if we foresee these values to be changeable at runtime, or alternatively to group them as constants. Otherwise it might look a bit overkill.

Reply:
Each CNV version may have different requirements. As far as I understand we need to get those values from the products and there is discussion in progress how to solve this. I can create something similar to OCS code but it seems not to be solution to the problem. @gamli75 What do you think?

Reply:
I think we'd better have them set as properly-named constants. I would like it be super-simple for a QE engineer (or hopefully, a script) to send a PR that updates CNV constants whenever a new CNV version is out.